frontend/src/features/contacts/contact/create/Person/CreatePersonForm.tsx

```diff
@@ -62,27 +67,51 @@ export const CreatePersonForm: React.FunctionComponent<ICreatePersonFormProps> =
   };
 
   const onSubmit = async (
-    values: ICreatePersonForm,
+    formPerson: ICreatePersonForm,
     { resetForm, setSubmitting }: FormikHelpers<ICreatePersonForm>,
   ) => {
     try {
-      await postPerson(personCreateFormToApiPerson(values));
-      setSubmitting(false);
-      toast.info('Contact added successfully');
-      history.push('/contact/list');
-    } catch (error) {
-      toast.error('Failed to add contact', { autoClose: 7000 });
+      setDuplicateModal(false);
+      let newPerson = personCreateFormToApiPerson(formPerson);
+      const personResponse = await addPerson(newPerson, setDuplicateModal, allowDuplicate);
+
+      console.log(personResponse);
+
+      if (!!personResponse?.id) {
+        history.push('/contact/list');
+      }
+    } finally {
+      formikRef?.current?.setSubmitting(false);
```

Comment:
you don't really need to use formikRef here (but is okay). The **onSubmit** functions receives all formik helpers as parameters - see function signature above